Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java

```diff
@@ -235,6 +263,114 @@ public void showTabs() {
         mViewPager.setPagingEnabled(true);
     }
 
+    public void setSearchMatchBarVisible(boolean isVisible) {
+        if (mSearchMatchBar != null) {
+            mSearchMatchBar.setVisibility(isVisible ? View.VISIBLE : View.GONE);
+        }
+    }
+
+    private void setUpSearchMatchBar(Intent intent) {
+        if (mSearchMatchIndexes == null) {
+            String matchOffsets = intent.getStringExtra(NoteEditorFragment.ARG_MATCH_OFFSETS);
+            String[] matches = matchOffsets != null ? matchOffsets.split("\\s+") : new String[]{};
+            String[] matchesStart = new String[matches.length / MATCH_INDEX_COUNT];
+
+            // Get "start" item from matches.  The format is four space-separated integers that
+            // represent the location of the match: "column token start length" ex: "1 3 3 7"
+            for (int i = MATCH_INDEX_START, j = 0; i < matches.length; i += MATCH_INDEX_COUNT, j++) {
+                matchesStart[j] = matches[i];
+            }
+
+            // Remove duplicate items with linked hash set and linked list.
+            matchesStart = new LinkedHashSet<>(new LinkedList<>(Arrays.asList(matchesStart))).toArray(new String[0]);
+            mSearchMatchIndexes = new int[matchesStart.length];
+
+            // Convert matches string array to integer array.
+            for (int i = 0; i < matchesStart.length; i++) {
+                mSearchMatchIndexes[i] = Integer.parseInt(matchesStart[i]);
+            }
+        }
+
+        mSearchMatchBar = findViewById(R.id.search_match_bar);
+        mNumberPosition = findViewById(R.id.text_position);
+        mNumberTotal = findViewById(R.id.text_total);
+        mButtonPrevious = findViewById(R.id.button_previous);
+        mButtonNext = findViewById(R.id.button_next);
+
+        mButtonPrevious.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSearchMatchIndex > 0) {
+                    mSearchMatchIndex--;
+                }
```

Comment:
if we were to flip this condition could not skip skip all the following updates if we're at the start of the list of matches (and unnecessary)…

```java
if (mSearchMatchIndex) {
  return;
}

mSearchIndex--;

mNoteEditorFragment.…
```

Reply:
Sorry, I don't follow.

```java
if (mSearchMatchIndex) {
  return;
}
```

`mSearchMatchIndex` is an integer rather than a boolean.

Reply:
that was meant to say `if (mSearchMatchIndex === 0) { return }`

sorry 'bout that

Reply:
Ah, okay.  No worries.  That makes more sense.

Incidentally, the code below the `if (mSearchMatchIndex > 0) { mSearchMatchIndex--; }` statement is not called when `mSearchMatchIndex <= 0` since the `updateSearchMatchBarStatus()` method disables the button and the click won't be fired.

We could update that snippet to the following.  Do you think that is more readable/understandable than the current code?

```java
if (mSearchMatchIndex > 0) {
    mSearchMatchIndex--;

    mNoteEditorFragment.scrollToMatch(mSearchMatchIndexes[mSearchMatchIndex]);
    new Handler().postDelayed(
        new Runnable() {
            @Override
            public void run() {
                updateSearchMatchBarStatus();
            }
        },
        getResources().getInteger(android.R.integer.config_mediumAnimTime)
    );
}
```

Reply:
I don't think it's more or less readable. it seems reasonable to provide this guard since we can't count on the fact that the button will be disabled. when it comes to style I'm flexible but I generally find that early-aborting and removing nesting makes the control flow simpler to follow and chart and less prone to bugs as a result of that - as a result of exiting the scope before we have more assignments/variables and conditions that we could forget to handle. it's not a blocker here and I know it could sound like I'm making a bigger deal out of this than needs be; I'm not trying to do that but rather just respond to the question "do you find this more understandable?"

Reply:
In cases when multiple `if`/`else if`/`else` statements are nested, I agree that aborting early is preferable.  In this case, there is only one `if` so the cognitive load is small.  I think you make a good point about being aware of the button enabled/disabled later.  I updated the conditional as I mentioned above to make is clear when the scroll occurs and the search match bar is updated in a34fa9ab.